File: CoreJava/account-abstract-app/src/com/techlab/account/Account.java
package com.techlab.account;

public abstract class Account {
	private int accNo;
	private String accountName;
	protected double balance;
	protected boolean transactionStatus = true;

	public Account(int accNo, String accountName, double balance) {
		this.accNo = accNo;
		this.accountName = accountName;
		this.balance = balance;
	}

	// getters
	public int getAccNo() {
		return accNo;
	}

	public String getAccountName() {
		return accountName;
	}

	public double getBalance() {
		return balance;
	}

	public boolean getTransactionStatus() {
		return transactionStatus;
	}

	public void deposit(int amount) {
		balance = balance + amount;
	}

	public abstract void withdraw(int amount);

}

File: C#Folder/CustomExceptionApp/CustomExceptionApp/Model/Account.cs
using System;

namespace CustomExceptionApp.Model
{
    class Account
    {
        private String _accountName;
        private double _balance;
        private static int _MINIMUM_BALANCE = 500;
        private bool _transactionStatus = true;

        public Account(String accountName, double balance)
        {
            this._accountName = accountName;
            this._balance = balance;
        }

        public string AccountName { get => _accountName; }

        public double Balance { get => _balance; }

        public static int MINIMUM_BALANCE { get => _MINIMUM_BALANCE; }

        public bool TransactionStatus { get => _transactionStatus; }

        public void Deposit(int amount)
        {
            _balance = _balance + amount;
            _transactionStatus = true;
        }

        public void Withdraw(int amount)
        {
            if ((_balance - amount) >= 500)
            {
                _balance = _balance - amount;
                _transactionStatus = true;
            }
            else
            {
                _transactionStatus = false;
            }
        }




    }
}

